util/src/main/java/io/grpc/util/AdvancedTlsX509TrustManager.java

```diff
@@ -43,30 +44,38 @@
 /**
  * AdvancedTlsX509TrustManager is an {@code X509ExtendedTrustManager} that allows users to configure
  * advanced TLS features, such as root certificate reloading, peer cert custom verification, etc.
+ * We expect only one of {@link AdvancedTlsX509TrustManager#useSystemDefaultTrustCerts()},
+ * {@link AdvancedTlsX509TrustManager#updateTrustCredentials(X509Certificate[])},
+ * {@link AdvancedTlsX509TrustManager#updateTrustCredentialsFromFile(File, long, TimeUnit,
+ * ScheduledExecutorService)},
+ * {@link AdvancedTlsX509TrustManager#updateTrustCredentialsFromFile(File)} methods to be called
+ * after instantiation of this class.
  * For Android users: this class is only supported in API level 24 and above.
  */
-@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8024")
 @IgnoreJRERequirement
 public final class AdvancedTlsX509TrustManager extends X509ExtendedTrustManager {
   private static final Logger log = Logger.getLogger(AdvancedTlsX509TrustManager.class.getName());
 
+  // Minimum allowed period for refreshing files with credential information.
+  private static final int MINIMUM_REFRESH_PERIOD = 1;
```

Comment:
nit: Prefer using `Duration` here.

Reply:
Same problem with the KeyManager - they use 'period' as a param name (plus it's long + TimeUnit) - 
https://github.com/grpc/grpc-java/blob/8aaace12eb8697a2dd841d2261b0051a1414f0c8/util/src/main/java/io/grpc/util/AdvancedTlsX509TrustManager.java#L222

Let's defer to Eric if I need to refactor all of it.

Reply:
We don't yet depend on java.time, as that requires API desugaring on Android.